assets/helpers/payPalExpressCheckout/payPalExpressCheckoutActions.js

```diff
@@ -0,0 +1,156 @@
+// @flow
+
+// ----- Imports ----- //
+
+import * as payPalExpressCheckout from './payPalExpressCheckout';
+
+
+// ----- Types ----- //
+
+export type Action =
+  | { type: 'START_PAYPAL_EXPRESS_CHECKOUT' }
+  | { type: 'SET_PAYPAL_EXPRESS_AMOUNT', amount: number }
+  | { type: 'PAYPAL_EXPRESS_CHECKOUT_LOADED' }
+  | { type: 'PAYPAL_EXPRESS_ERROR', message: string }
+  ;
+
+
+// ----- Actions ----- //
+
+function startPayPalExpressCheckout(): Action {
+  return { type: 'START_PAYPAL_EXPRESS_CHECKOUT' };
+}
+
+export function setPayPalExpressAmount(amount: number): Action {
+  return { type: 'SET_PAYPAL_EXPRESS_AMOUNT', amount };
+}
+
+function payPalExpressCheckoutLoaded(): Action {
+  return { type: 'PAYPAL_EXPRESS_CHECKOUT_LOADED' };
+}
+
+export function payPalExpressError(message: string): Action {
+  return { type: 'PAYPAL_EXPRESS_ERROR', message };
+}
+
+function handleSetupResponse(response: Object) {
+  let resp = null;
+  if (response.status === 200) {
+    resp = response.json();
+  }
+
+  return resp;
+}
+
+// First Step: setupPayment
+// Sends request to server to setup payment, and returns Paypal token.
+function setupPayment(dispatch, state: Object) {
+
+  return (resolve, reject) => {
+    const requestBody = {
+      amount: state.payPalExpressCheckout.amount,
+      billingPeriod: state.payPalExpressCheckout.billingPeriod,
+      currency: state.payPalExpressCheckout.currency,
+    };
+    const SETUP_PAYMENT_URL = '/paypal/setup-payment';
+
+    fetch(SETUP_PAYMENT_URL, {
+      credentials: 'include',
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json', 'Csrf-Token': state.csrf.token },
+      body: JSON.stringify(requestBody),
+    }).then(handleSetupResponse)
+      .then((token) => {
+        if (token) {
+          resolve(token.token);
+        } else {
+          dispatch(payPalExpressError('PayPal token came back blank.'));
+        }
+      }).catch((err) => {
+        dispatch(payPalExpressError(err.message));
+        reject(err);
+      });
+  };
+}
+
+// Second step: createAgreement
+function createAgreement(payPalData, state: Object) {
+  const CREATE_AGREEMENT_URL = '/paypal/create-agreement';
+  return fetch(CREATE_AGREEMENT_URL, {
+    credentials: 'include',
+    headers: { 'Content-Type': 'application/json', 'Csrf-Token': state.csrf.token },
+    method: 'POST',
+    body: JSON.stringify({ token: payPalData.paymentToken }),
+  }).then(response => response.json());
```

Comment:
Yes, I totally agree that we have to refactor this chunk of code. The purpose of this PR is to test the PayPal integration in PROD. I am planning to refactor `ajax.js` and merge the two file together. I didn't want to add this logic right now to that file because it would crash with the one-off logic.